docs/Contributing-back.rst

```diff
@@ -0,0 +1,496 @@
+
+.. code:: shell
+
+  git remote add myfork git@github.com:EXAMPLE/easybuild-framework.git
+
+.. note:: **Do not forget to replace** '``EXAMPLE``' **with your GitHub account name.**
+
+          '``myfork``' is just a name that you give to the remote for your
+          fork of the repository on GitHub; you can modify this to your liking
+          if desired (but do take that into account for the further instructions if you do so).
+
+          For the central repository, a default remote named ``origin`` should
+          have been created via ``git clone`` (see the output of ``git remote -v``).
+
+Finally, it is recommended to also check out the ``develop`` branch,
+which will be used as the base for your own branches:
+
+.. code:: shell
+
+  git checkout -b develop origin/develop
+
+With this in place, you are all set to open pull requests for your contributions.
+
+.. _contributing_back_creating_pull_requests:
+
+Opening a new pull request
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note:: This section describes the manual procedure to open a new pull request.
+
+          Please consider using ``eb --new-pr`` instead, see :ref:`github_new_pr`.
+
+.. note:: We are assuming you are already located in your local working copy of
+          the repository you want to contribute to (e.g., ``easybuild-framework``).
+
+To open a pull request for your contribution, you must follow these steps:
+
+i. :ref:`contributing_back_creating_pull_requests_branch`
+ii. :ref:`contributing_back_creating_pull_requests_commit`
+iii. :ref:`contributing_back_creating_pull_requests_push`
+iv. :ref:`contributing_back_creating_pull_requests_open_pr`
+
+
+.. _contributing_back_creating_pull_requests_branch:
+
+Creating a new branch
++++++++++++++++++++++
+
+First, create a new branch for your work. You can do this either before
+or after making the changes that you would like to contribute back, but we
+recommend to create a new branch before making any changes.
+
+Make sure you have the ``develop`` branch checked out before creating your branch:
+
+.. code:: shell
+
+  git checkout develop
+
+To create a new branch, you should use ``git branch <branch_name>`` followed by
+``git checkout <branch_name>``, or equivalently ``git checkout -b <branch_name>``.
+
+For example, to create a new branch named ``mybranch``:
+
+.. code:: shell
+
+  git checkout -b mybranch
+
+You can choose the branch name freely, but make it sufficiently descriptive --
+your future self will thank you.
+
+
+.. _contributing_back_creating_pull_requests_commit:
+
+Committing your changes
++++++++++++++++++++++++
+
+To 'save' your changes, you should create one or more *commits* in the branch
+you created. It is recommended to make separate commits for different 'units of work'.
+
+First, make sure you have checked out the branch were you want to commit the changes to.
+
+For example, to commit changes to the ``mybranch`` branch:
+
+.. code:: shell
+
+  git checkout mybranch
```

Comment:
no, this is the preparation for committing to the `mybranch` branch; I'll clarify that